main.py

```diff
@@ -0,0 +1,221 @@
+import os
+import sys
+import json
+import subprocess
+
+folder_executed = sys.argv[1]
+
+def install_modules():
+    required_modules = [
+        'requests',
+        'natsort',
+        'Pillow',
+        'tqdm',
+        'asyncio',
+        'packaging',
+        'flask',
+    ]
+    
+    for module in required_modules:
+        try:
+            if module == 'Pillow':
+                __import__('PIL')
+            
+            else:
+                __import__(module)
+                
+        except ImportError:
+            subprocess.run(['pip', 'install', module])
+            
+    os.system('cls') if os.name == 'nt' else os.system('clear')
+install_modules()
+
+import requests
+
+
+temp_folder = os.environ['TEMP']
+app_folder = os.path.join(temp_folder, "MangaDex Uploader (APP)")
+path = os.path.join(temp_folder, "MangaDex Uploader (APP)", "run.py")
+config_path = os.path.join(app_folder, 'config.json')
+map_path_file = os.path.join(app_folder, 'name_id_map.json')
+
+os.makedirs(app_folder, exist_ok=True)
+os.makedirs(os.path.join(app_folder, "to_upload"), exist_ok=True)
+os.makedirs(os.path.join(app_folder, "uploaded"), exist_ok=True)
+
+
+folders = ['logs', 'static', 'templates', 'mupl\\http', 'mupl\\loc', 'mupl\\uploader', 'mupl\\utils']
+
+for x in folders:
+    x = os.path.join(app_folder, x)
+    os.makedirs(x, exist_ok=True)
+
+context_1 = 'https://raw.githubusercontent.com/OneDefauter/mupl/main'
```

Comment:
There are also better ways to have this downloaded. Instead of defining each directory separately and making each folder separately and then defining the folder structure separately (which can break easily) just use the GitHub api to find which path is a directory and which path is a file.

Look at [updater.py](/mupl/updater.py) in this repo for extracting the GitHub release zipball or look at the [publoader updater](https://github.com/ArdaxHz/publoader/blob/master/publoader/updater.py) for downloading and installing the latest commit changes without defining specific directories or files.